core/src/main/java/org/infinispan/stats/impl/CacheContainerStatsImpl.java

```diff
@@ -29,18 +28,12 @@
    public CacheContainerStatsImpl(EmbeddedCacheManager cm) {
       this.cm = cm;
       cm.getGlobalComponentRegistry().registerComponent(this, CacheContainerStats.class);
```

Comment:
This line is not right. You should instead add `@Scope(Scopes.GLOBAL)` in CacheContainerStats class.